File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.py
class Solution(object):
    def largestGoodInteger(self, num):
        """
        :type num: str
        :rtype: str
        """
        sub = ""
        n = len(num)

        for i in range(n - 2):
            if num[i] == num[i+1] == num[i+2]:
                temp = num[i:i+3]
                if  temp > sub:
                    sub = temp
        return sub

File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string s) {
        string sub = "";
        int n = s.size();

        for (int i = 0; i <= n - 3; i++) {
            if (s[i] == s[i+1] && s[i] == s[i+2]) {
                string temp = s.substr(i, 3);
                if (sub.empty() || temp > sub) {
                    sub = temp;
                }
            }
        }
        return sub;
    }
};